src/core/index.js

```diff
@@ -30,34 +31,49 @@ const getDefaultOptions = () => ({
       '/dns4/node0.preload.ipfs.io/https',
       '/dns4/node1.preload.ipfs.io/https'
     ]
-  }
+  },
+  repoAutoMigrate: false
 })
 
 async function create (options) {
   options = mergeOptions(getDefaultOptions(), options)
 
   // eslint-disable-next-line no-console
   const print = options.silent ? log : console.log
+  const repo = typeof options.repo === 'string' || options.repo == null
+    ? createRepo({ path: options.repo, autoMigrate: options.repoAutoMigrate })
+    : options.repo
 
   const apiManager = new ApiManager()
-
   const { api } = apiManager.update({
-    init: Components.init({ apiManager, print, options }),
+    init: Components.init({
+      apiManager,
+      print,
+      options,
+      repo
+    }),
     dns: Components.dns(),
-    isOnline: Components.isOnline({})
+    isOnline: Components.isOnline({}),
+    repo
```

Comment:
This is not the repo API from https://github.com/ipfs/js-ipfs/tree/master/src/core/components/repo